### ISSUE TITLE
Shrink _recipe.py

### ISSUE THREAD

**Issue by valeriupredoi** [ESMValGroup/ESMValCore#639]
I think we have reached the size and complexity limit for `_recipe.py` (we may have reached that a while ago) so I reckon we should start shrinking it and optimizing its use; I reckon the first step is to move all the preprocessor module specific functions (like `_update_{STEP}_settings`) to the preprocessor itself. What say you @bouweandela - if you agree I can get cracking with it starting now :beer:

**Comment by bouweandela** [comment]
Hi V,

I agree with you that it would be good to clean this file up. In the next few months we will be working on #498. I expect that once we have a more complete Python API in place, _recipe.py will also be a lot simpler, so I think it would be best to wait a bit with this.

Another issue is that at the moment we have about 90 open pull requests (45 in ESMValCore and 46 in ESMValTool), I would really like to focus on getting those merged before starting new work.

**Comment by valeriupredoi** [comment]
makes sense, man! let's wait a bit more, I'll dedicate my effeorts towards the noble cause of solving as many of those PR's and issues :carousel_horse:
